File: python/factor_count.py
from math import isqrt
def main():
    while True:
        potential = input("Please input a number to check for prime or press 'q' to quit: ")
        factors = []
        if potential.lower() == "q":
            quit()
        try:
            potential_num =int(potential)
        except ValueError:
            print("Please input a valid positive integer (number > 0)")
        else:
            if potential_num < 1:
                print("This number cannot be calculated for factors due to being smaller than one.")
            else:
                for number in range(1,isqrt(potential_num)+1):
                    if potential_num % number == 0:
                        factors.append(number)
                        factors.append(potential_num//number)
                factors = sorted(list(set(factors)))
                print(f"The factors of {potential_num} are: {factors}")
                print(f"The number is {'not ' if len(factors) != 2 else ''}prime, because it has {len(factors)} factor(s).")
if __name__ == "__main__":
    main()

File: csharp/factor_count.cs
List<ulong> factor_calculate(ulong number){
    List<ulong> factor_values= new List<ulong>{};
    for (ulong i =1;i<Math.Round(Math.Sqrt(number));i++){
        if (number % i == 0){
            factor_values.Add(i);
            factor_values.Add(number/i);
        }
    }
    if (Math.Floor(Math.Sqrt(number)) == Math.Sqrt(number)){
        factor_values.Add((ulong)Math.Floor(Math.Sqrt(number)));
    }
    factor_values.Sort();
    return factor_values;
}
void main(){
    int EXIT_SUCCESS=0;
    while (true){
            bool is_valid_input=false;
            ulong real_number=0;
            string? number_input;
            while (!is_valid_input){
                Console.Write("Insert a number to check its factors or press 'q' to quit: ");
                number_input=Console.ReadLine();
                try{
                    if(number_input.Equals("q", StringComparison.CurrentCultureIgnoreCase)){
                            Environment.Exit(EXIT_SUCCESS);
                    }
                    real_number= Convert.ToUInt64(number_input);
                }
                catch(FormatException){
                    Console.WriteLine("This input is invalid because it is not a valid number!");
                    continue;
                }
                catch (OverflowException){
                    Console.WriteLine("This input is not valid because the number is outside the bounds of this program");
                    continue;
                }
                is_valid_input=true;
            }
            List<ulong> factors=factor_calculate(real_number);
            Console.WriteLine($"Factors of {real_number}: [{string.Join(",",factors)}]");

            if (real_number<2){
                Console.WriteLine($"{real_number} is not prime because it is smaller than 2.");
                }
            Console.WriteLine($"{real_number} is {(factors.Count!=2?"not ":"")}prime because it has {factors.Count} factors.");
        }
}

main();